packages/eas-cli/src/submissions/utils/builds.ts

```diff
@@ -1,11 +1,16 @@
-import { AppPlatform, BuildStatus } from '../../graphql/generated';
+import { AppPlatform, BuildFragment, BuildStatus } from '../../graphql/generated';
 import { BuildQuery } from '../../graphql/queries/BuildQuery';
 
-export async function getBuildArtifactUrlByIdAsync(
+export type SubmittedBuildInfo = {
+  buildId: BuildFragment['id'];
+  artifactUrl: string;
+} & Pick<BuildFragment, 'createdAt' | 'appVersion' | 'appBuildVersion' | 'platform'>;
+
+export async function getBuildInfoByIdAsync(
   platform: AppPlatform,
   buildId: string
-): Promise<string> {
-  const { platform: buildPlatform, artifacts } = await BuildQuery.byIdAsync(buildId);
+): Promise<SubmittedBuildInfo> {
```

Comment:
I'd make this function (and the other - `getLatestBuildInfoAsync`) return `BuildFragment` instead. Since the full data is already fetched from the server it doesn't make sense to return a limited set of fields.
If you apply this comment, then you can get rid of `SubmittedBuildInfo`.